lib/mayaUsd/nodes/layerManager.cpp

```diff
@@ -555,6 +577,12 @@ bool LayerDatabase::saveUsd(bool isExport)
             return true;
         }
 
+        // After the potentially partial save, we need to refresh the stages
+        // to be saved because the saving might have modified the proxy shape
+        // attributes and we need to re-evaluate these nodes so that the stages
+        // are re-created with the new attribute values if needed.
+        refreshProxiesToSave();
```

Comment:
By doing this pull here does it cause any notifications to be sent and could that potentially mess up the clients since we've only done a partial save?

Reply:
At the point this is called, we are in a pre-save callback AFAIK, so there should be no issue.